src/views/editor/index.vue

```diff
@@ -392,47 +434,24 @@ export default {
      * @return {Promise<*>}
      */
     async deploy() {
-      /**
-       * Check if deploy function is available
-       */
-      if (typeof this.compiled.deploy !== 'function') {
+      if (typeof this.instance.deploy !== 'function') {
```

Comment:
You always have this function in contract instance and also deploy function will compile contract itself if needed

Reply:
Updating.

Reply:
Though, I still need to check if the function is there or not in either cases, because the deploy step is separate from compile step. The user first needs to compile and then deploy. The contract instance gets only created on the compile step.

Reply:
I'll refactor the `editor` in another round (where we'll add some extra functionalities and change the UI, since the sidebar configuration is a bit difficult to manage), but for now this is good.